pkg/printer/printer.go

```diff
@@ -384,6 +384,170 @@ func (p *Printer) WriteProtoNodeStatusEvent(r *observerpb.GetFlowsResponse) erro
 	return nil
 }
 
+func formatServiceAddr(a *pb.ServiceUpsertNotificationAddr) string {
+	return fmt.Sprintf("%s:%d", a.Ip, a.Port)
+}
+
+func getAgentEventDetails(e *pb.AgentEvent) string {
+	switch e.GetType() {
+	case pb.AgentEventType_AGENT_EVENT_UNKNOWN:
+		if u := e.GetUnknown(); u != nil {
+			return fmt.Sprintf("type: %s, notification: %s", u.Type, u.Notification)
+		}
+	case pb.AgentEventType_AGENT_STARTED:
+		if a := e.GetAgentStart(); a != nil {
+			return fmt.Sprintf("start time: %s", fmtTimestamp(a.Time))
+		}
+	case pb.AgentEventType_POLICY_UPDATED, pb.AgentEventType_POLICY_DELETED:
+		if p := e.GetPolicyUpdate(); p != nil {
+			return fmt.Sprintf("labels: [%s], revision: %d, count: %d",
+				strings.Join(p.Labels, ","), p.Revision, p.RuleCount)
+		}
+	case pb.AgentEventType_ENDPOINT_REGENERATE_SUCCESS, pb.AgentEventType_ENDPOINT_REGENERATE_FAILURE:
+		if r := e.GetEndpointRegenerate(); r != nil {
+			var sb strings.Builder
+			fmt.Fprintf(&sb, "id: %d, labels: [%s]", r.Id, strings.Join(r.Labels, ","))
+			if re := r.Error; re != "" {
+				fmt.Fprintf(&sb, ", error: %s", re)
+			}
+			return sb.String()
+		}
+	case pb.AgentEventType_ENDPOINT_CREATED, pb.AgentEventType_ENDPOINT_DELETED:
+		if ep := e.GetEndpointUpdate(); ep != nil {
+			var sb strings.Builder
+			fmt.Fprintf(&sb, "id: %d", ep.Id)
+			if n := ep.PodName; n != "" {
+				fmt.Fprintf(&sb, ", pod name: %s", n)
+			}
+			if n := ep.Namespace; n != "" {
+				fmt.Fprintf(&sb, ", namespace: %s", n)
+			}
```

Comment:
It feels a bit counterintuitive to have the pod name before the namespace. Would you mind swapping the order?

Reply:
Agree. I assume same goes for `pb.AgentEventType_SERVICE_UPSERTED` case below where we want the service name before the namespace?

Reply:
Forgot to amend this change as well :facepalm: now addressed in #454